cephfs/path_test.go

```diff
@@ -60,3 +60,273 @@ func TestRemoveDir(t *testing.T) {
 	assert.EqualError(t, err,
 		fmt.Sprintf("stat %s: no such file or directory", localPath))
 }
+
+func TestLink(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("directory operations", func(t *testing.T) {
+		// Root dir, both as source and destination.
+		err := mount.Link("/", "/")
+		// Error directory operations are not allowed.
+		assert.Error(t, err)
+
+		dirname := "mydir"
+		localPath := path.Join(CephMountDir, dirname)
+		err = mount.MakeDir(dirname, 0755)
+		assert.NoError(t, err)
+		_, err = os.Stat(localPath)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, mount.RemoveDir(dirname))
+		}()
+
+		// Creating link for a directory.
+		err = mount.Link(dirname, "/")
+		// Error, directory operations not allowed.
+		assert.Error(t, err)
+
+		filename := "anyFile.txt"
+		err = mount.Link(dirname, filename)
+		// Error, directory operations not allowed.
+		assert.Error(t, err)
+
+		err = mount.Link(filename, dirname)
+		// Error, file does not exist.
+		assert.Error(t, err)
+
+		fname := "TestFile.txt"
```

Comment:
This sub-section should probably be it's own sub-test (with t.Run).

Reply:
Done